skipInitiBuildIndex-multibranch.groovy

```diff
@@ -0,0 +1,48 @@
+/**
+ * This script is used to enable the build strategy SkipInitialBuildOnFirstIndexingResetRevision for all multibranch projects that have NOT set a Build Strategy definition.
+ * Use Case: Migration of Jobs across CloudBees CI instances to prevent Build Storm.
+ * Requirement: https://docs.cloudbees.com/docs/release-notes/latest/plugins/cloudbees-build-strategies-plugin/
+ * Tested on: CloudBees CI 2.462.3.3 
+ */
+
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject
+import com.cloudbees.jenkins.plugins.buildstrategies.SkipInitialBuildOnFirstIndexingResetRevision
+
+def dryRun = true
+
+
+def enableBuildStrategy(dryRun) {
+    def modifiedCount = 0
+    def emptyCount = 0
+    def nonEmptyCount = 0
+
+    Jenkins.instance.getAllItems(WorkflowMultiBranchProject).each { multibranchProject ->
+		
+        multibranchProject.getSourcesList().each { branchSource ->
+            if (branchSource !=null) { 
+                def buildStrategies = branchSource.getBuildStrategies()
+                if (buildStrategies.isEmpty()) {
+                    if (dryRun){
+                        println "Dryrun: ${multibranchProject.fullName} is a candidate to add Build Strategy."
+                    } else {
+                        branchSource.setBuildStrategies(Arrays.asList(new SkipInitialBuildOnFirstIndexingResetRevision()))
+                        println "Adding Build Strategy SkipInitialBuildOnFirstIndexingResetRevision for ${multibranchProject.fullName}" 
+                        modifiedCount++
+                    }
+                    emptyCount++
+                } else {
+                    println "${multibranchProject.fullName} has already enabled Build Strategy ${buildStrategies}."
+                    nonEmptyCount++
+                }
+            }
+        }
+    }
+    println "===================================================================================="
+    println "Non-empty build strategies count: ${nonEmptyCount}"
+    println "Empty build strategies count: ${emptyCount}"
+    println "Added build strategies SkipInitialBuildOnFirstIndexingResetRevision: ${modifiedCount}"
+    println "===================================================================================="
+}
+
+enableBuildStrategy(dryRun)
```

Comment:
```suggestion
try (ACLContext ctx = ACL.as2(ACL.SYSTEM2)) {
    enableBuildStrategy(dryRun)
}
```

When script running through Script Conse this should avoid lookups to RBAC and the security realm, afaik.

Reply:
I tried this but I am getting this error

```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
Script1.groovy: 74: expecting '{', found '(' @ line 74, column 5.
   try (ACLContext ctx = ACL.as2(ACL.SYSTEM2)) {
       ^
1 error
```

Looking into [rbac plugin snippets examples](https://github.com/search?q=repo%3Acloudbees%2Fnectar-rbac-plugin%20ACLContext&type=code) seems correct but maybe is not valid in the groovy context.

Any ideas?

Reply:
It is most likely not a valid groovy syntax, since I copy pasted this from Java.

Reply:
I think we should keep it as is. Switching to SYSTEM might make it faster, but it also loses the auditing.